packages/flame/lib/src/components/position_component.dart

```diff
@@ -332,6 +343,23 @@ class PositionComponent extends Component
   /// The absolute center of the component.
   Vector2 get absoluteCenter => absolutePositionOfAnchor(Anchor.center);
 
+  /// Returns the angle formed by component's forward vector and a vector
+  /// starting at component's absolute position and ending at target.
+  /// Uses [nativeAngle] to decide the forward direction of the component.
+  /// See [lookAt] to make the component instantly rotate towards target.
+  /// Note: Returns 0 if target is exactly same as [absolutePosition].
+  double angleTo(Vector2 target) {
+    return math.atan2(
+          target.x - absolutePosition.x,
+          absolutePosition.y - target.y,
+        ) -
```

Comment:
A bit unclear about the choice of coordinate systems here: why is the current component uses `absolutePosition`, but the target has simple position, shouldn't it use the absolute position as well?

Another concern is the rotation of the parent coordinate frame: we may need to add something like `absoluteAngle - angle` here.

Reply:
> why is the current component uses `absolutePosition`, but the target has simple position, shouldn't it use the absolute position as well

Yes, `target` also need to be in absolute coordinate system. But it is just a `Vector2`, so can't really enforce users to pass in absolutePosition. It is similar to how `camera.followVector2()` works. If users supply local position of a component it probably won't work as expected.

Should I change it to pass in a position component so that we can internally get the `absolutePosition`?

Reply:
> Another concern is the rotation of the parent coordinate frame: we may need to add something like `absoluteAngle - angle` here.

Fixed this and added test for nested components as well.

Reply:
> Should I change it to pass in a position component so that we can internally get the absolutePosition?

I think that it should be a vector2 as the argument just like it is, but we should specify in the docs that it should be an absolute vector.

Reply:
Added comment regarding `target`'s coordinate system.